lib/internal/process/next_tick.js

```diff
@@ -210,45 +251,48 @@ function setupNextTick() {
     } while (tickInfo[kLength] !== 0);
   }
 
-  function TickObject(callback, args, domain) {
-    this.callback = callback;
-    this.domain = domain;
-    this.args = args;
-    this[async_id_symbol] = -1;
-    this[trigger_id_symbol] = -1;
-  }
-
-  function setupInit(tickObject, triggerId) {
-    tickObject[async_id_symbol] = ++async_uid_fields[kAsyncUidCntr];
-    tickObject[trigger_id_symbol] = triggerId || initTriggerId();
-    if (async_hook_fields[kInit] > 0) {
-      emitInit(tickObject[async_id_symbol],
-               'TickObject',
-               tickObject[trigger_id_symbol],
-               tickObject);
+  class TickObject {
+    constructor(callback, args, asyncId, triggerId) {
+      this.callback = callback;
+      this.args = args;
+      this.domain = process.domain || null;
+      this[async_id_symbol] = asyncId;
+      this[trigger_id_symbol] = triggerId;
     }
   }
 
+  // `nextTick()` will not enqueue any callback when the process is about to
+  // exit since the callback would not have a chance to be executed.
   function nextTick(callback) {
     if (typeof callback !== 'function')
       throw new errors.TypeError('ERR_INVALID_CALLBACK');
-    // on the way out, don't bother. it won't get fired anyway.
+
     if (process._exiting)
       return;
 
     var args;
-    if (arguments.length > 1) {
-      args = new Array(arguments.length - 1);
-      for (var i = 1; i < arguments.length; i++)
-        args[i - 1] = arguments[i];
+    switch (arguments.length) {
+      case 1: break;
+      case 2: args = [arguments[1]]; break;
+      case 3: args = [arguments[1], arguments[2]]; break;
+      case 4: args = [arguments[1], arguments[2], arguments[3]]; break;
+      default:
+        args = new Array(arguments.length - 1);
+        for (var i = 1; i < arguments.length; i++)
+          args[i - 1] = arguments[i];
```

Comment:
I'm not sure how spread could be used here. Do you mean rest parameters? If so, I tried that and while it's a *little* faster in one case, it's slower to varying degrees in all others.